benchmark848/benchmark.cpp

```diff
@@ -0,0 +1,348 @@
+    CreateTable(connection, nextTableName);
+    Checkpoint(connection);
+
+    /* 3. Drop the second tables here to wrap up the test */
+    DropTable(connection, nextTableName);
+    Checkpoint(connection);
+}
+
+/*
+ * Alter Table Test
+ * This test is limited to test Alter Table DROP ... for now
+ */
+void AlterTableTest(const unique_ptr<Connection> &connection)
+{
+    string tableName = RandomSelectTable("");
+    int tableIndex = FindTableIndex(tableName);
+    string nextTableName = RandomSelectTable(tableName);
+
+    /*
+     * 1. Create table here first
+     *      Checkpoint will create physical storage
+     */
+    CreateTable(connection, tableName);
+    Checkpoint(connection);
+
+    /*
+     * 2. Alter the table and Create another table.
+     *      chunks are recycled in ALTER.
+     *      Checkpoint will create physical storage for new table
+     */
+    int numColumn = tableNumColumns[tableIndex];
+    int droppedColIndex = rand() % numColumn;
+    string dropColName = tableColumns[tableIndex][droppedColIndex];
+    AlterTable(connection, tableName, DROP_COLUMN, dropColName, INVALID_COLUMN_TYPE);
+    CreateTable(connection, nextTableName);
+    Checkpoint(connection);
+
+    /* 3. Drop both tables here to wrap up the test */
+    DropTable(connection, tableName);
+    DropTable(connection, nextTableName);
+    Checkpoint(connection);
+}
+
+void displayQueryResult(const unique_ptr<QueryResult> &result)
+{
+    while (result->hasNext()) {
+        cout << "| ";
+        const auto row = result->getNext();
+        if (row->len() == 3) {
+            cout << row->getValue(0)->getValue<string>() << " | "
+                << row->getValue(1)->getValue<int64_t>() << " | "
+                << row->getValue(2)->getValue<int64_t>() << " | " << endl;
+        } else {
+            cout << row->getValue(0)->getValue<string>() << " | ";
+        }
+    }
+}
+
+/* Delete Node Group Test */
+void DeleteNodeGroupTest(const unique_ptr<Connection> &connection) {
+    string tableName = RandomSelectTable("");
+    string nextTableName = RandomSelectTable(tableName);
+
+    /*
+     * 1. Create table here first
+     *      Checkpoint will create physical storage
+     */
+    CreateTable(connection, tableName);
+    Checkpoint(connection);
+
+    /*
+     * 2. Delete table entries based on random ratio
+     *      chunks are recycled in Checkpoint.
+     */
+    int beginId = (rand() % 50) * NUM_ROWS / 100;
+    int endId = (rand() % 50 + 50) * NUM_ROWS / 100;
+    DeleteEntries(connection, tableName, beginId, endId);
+    Checkpoint(connection);
+
+    /*
+     * 3. Create a new table to resue recycled spaces
+     */
+    CreateTable(connection, nextTableName);
+    Checkpoint(connection);
+
+    /* 4. Drop both tables here to wrap up the test */
+    DropTable(connection, tableName);
+    DropTable(connection, nextTableName);
+    Checkpoint(connection);
+}
+
+int main(int argc, char* argv[])
+{
+    if (argc == 1) {
```

Comment:
Since we are expecting at least 3 arguments, the condition may be modified accordingly.

Reply:
4 now with the updated code

Reply:
Fixed. We actually need 7 since argc will also count "-N/-B/-D"